lib/gems/pending/util/miq-process.rb

```diff
@@ -6,19 +6,12 @@
 require 'util/miq-system'
 
 class MiqProcess
+  # Collect and return a list of PID's for all processes that match
+  # +process_name+ or +process_name.exe+.
+  #
   def self.get_active_process_by_name(process_name)
-    pids = []
-
-    case Sys::Platform::IMPL
-    when :mswin, :mingw
-      require 'util/win32/miq-wmi'
-      WMIHelper.connectServer.run_query("select Handle,Name from Win32_Process where Name = '#{process_name}.exe'") { |p| pids << p.Handle.to_i }
-    when :linux, :macosx
-      pids = `ps -e | grep #{process_name} | grep -v grep `.split("\n").collect(&:to_i)
-    else
-      raise "Method MiqProcess.get_active_process_by_name not implemented on this platform [#{Sys::Platform::IMPL}]"
-    end
-    pids
+    procs = Sys::ProcTable.ps(:smaps => false, :cgroup => false)
```

Comment:
I forgot that you did add the kwargs stuff, so this is great.

Looking at the `sys-proctable` code, does seem like there are a few extra files being read and "Dir's glob'd" that aren't needed for this use case, but that is something to improved on in the future.

Reply:
@NickLaMuro Yeah, it's a bit less efficient, but not terribly so with the options being passed.